<launch>

  <arg name="save_folder" />
  <arg name="image_extension"           default="png" />
  <arg name="starting_index"            default="1" />
  <arg name="image_filename"            default="image_" />
  <arg name="cloud_filename"            default="cloud_" />

  <arg name="search_checkerboard"       default="true" />
  <arg name="rate"                      default="2.0" />
  
  <arg name="camera_type"               default="pinhole" />
  <arg name="kinect_name"               default="kinect" />
  
  <include file="$(find openni2_launch)/launch/openni2.launch">
    <arg name="camera"                value="$(arg kinect_name)" />
    <arg name="depth_registration"    value="false" />
    <arg name="publish_tf"            value="false" />
  </include>

  <node pkg="rgbd_calibration" type="rgbd_data_collection" name="rgbd_data_collection" output="screen" required="true">
    <param name="save_folder"             value="$(arg save_folder)" />
    <param name="image_extension"         value="$(arg image_extension)" />
    <param name="starting_index"          value="$(arg starting_index)" />
    <param name="image_filename"          value="$(arg image_filename)" />
    <param name="cloud_filename"          value="$(arg cloud_filename)" />
    <param name="camera_type"             value="$(arg camera_type)" />
    <param name="search_checkerboard"     value="$(arg search_checkerboard)" />
    
    <remap from="~image"                  to="/$(arg kinect_name)/rgb/image_rect_color" />
    <remap from="~camera_info"            to="/$(arg kinect_name)/rgb/camera_info" />
    <remap from="~point_cloud"            to="/$(arg kinect_name)/depth/points" />
  </node>

  <node if="$(arg search_checkerboard)"
        pkg="rostopic" type="rostopic" name="checkerboards_pub" output="screen" 
        args="pub -f $(find rgbd_calibration)/conf/checkerboards.yaml rgbd_data_collection/checkerboard_array
              calibration_msgs/CheckerboardArray --latch">
  </node>

</launch>
